__tests__/screens/CredentialOffer.test.tsx

```diff
@@ -1,126 +1,129 @@
+    const tree = render(<CredentialOffer route={props as any} navigation={useNavigation()} />)
+    const declineBtn = await tree.findByText('Global.Decline')
+
+    fireEvent(declineBtn, 'press')
+
+    expect(spyAlert).toHaveBeenCalledTimes(1)
+
+    await waitFor(async () => {
+      // @ts-ignore
+      await spyAlert.mock.calls[0][2][1].onPress()
     })
 
-    /**
-     * Given the holder accepts a credential offer from an issuer they are connected with
-     * When the credential arrives in the wallet
-     * Then the screen will change from the loading screen to a success screen informing the holder that the credential has arrived
-     */
-    test('a success screen is displayed when the credential arrives', async () => {
-      testCredentialRecords[0].state = CredentialState.CredentialReceived
+    expect(tree).toMatchSnapshot()
+  })
 
-      const { findByText, UNSAFE_getByProps } = render(
-        <CredentialOffer
-          navigation={useNavigation()}
-          route={{ params: { credentialId: testCredentialRecords[0].id } } as any}
-        />
-      )
+  test('shows credential delivery modal', async () => {
+    // @ts-ignore
+    useCredentialById.mockReturnValue(credentialRecord)
+    const tree = render(<CredentialOffer route={props as any} navigation={useNavigation()} />)
 
-      const acceptButtonInstance = await findByText('Global.Accept')
+    const acceptBtn = await tree.findByText('Global.Accept')
+    fireEvent(acceptBtn, 'press')
 
-      fireEvent(acceptButtonInstance, 'press')
+    const onTheWayModal = tree.getByTestId('CredentialOffer.CredentialOnTheWay')
+    const declinedModal = tree.getByTestId('CredentialOffer.CredentialDeclined')
+    const addedModal = tree.getByTestId('CredentialOffer.CredentialAddedToYourWallet')
 
-      const notificationModalInstance = await UNSAFE_getByProps({
-        title: 'CredentialOffer.CredentialAddedToYourWallet',
-      })
+    expect(onTheWayModal.props.visible).toBeTruthy()
+    expect(declinedModal.props.visible).toBeFalsy()
+    expect(addedModal.props.visible).toBeFalsy()
+  })
 
-      expect(notificationModalInstance).not.toBe(null)
-      expect(notificationModalInstance.type).toBe(NotificationModal)
-      expect(notificationModalInstance.props.visible).toBeTruthy()
-    })
+  test('handle declined credential', async () => {
+    const myCredential = new CredentialRecord(credential)
+    myCredential.createdAt = new Date(credentialRecord.createdAt)
+    myCredential.state = CredentialState.Declined
+    // @ts-ignore
+    useCredentialById.mockReturnValue(myCredential)
+
+    const tree = render(<CredentialOffer route={props as any} navigation={useNavigation()} />)
+
+    const onTheWayModal = tree.getByTestId('CredentialOffer.CredentialOnTheWay')
+    const declinedModal = tree.getByTestId('CredentialOffer.CredentialDeclined')
+    const addedModal = tree.getByTestId('CredentialOffer.CredentialAddedToYourWallet')
 
-    /**
```

Comment:
I usually like to keep the acceptance criteria commented so we can see how the tests are structured to conform to those.

Reply:
Ok. I'll see what I can add back in. My though on unit tests is they should run fast and be easy to maintain. I generally run `npm run test:watch` and am shooting for close to 80 code coverage including any key execution paths. Some times that won't necessarily align with acceptance criteria. Also, a snapshot test covers lots of ground and may cover one or more acceptance criteria (lots of bang for your buck).

Reply:
I agree on snapshots for testing presentation layout. I still think we should have functional tests that conform to the acceptance criteria that cover functional requirements. Ultimately that's what ensures conformity when the app is iterated on.

Reply:
Ok. I don't want to add them to my tests. Its just more work and makes them harder to maintain.